smartsim/_core/launcher/colocated.py

```diff
@@ -113,26 +107,45 @@ def _build_colocated_wrapper_cmd(
         sys.executable,
         "-m",
         "smartsim._core.entrypoints.colocated",
-        "+ifname",
-        interface,
         "+lockfile",
         lockfile,
         "+db_cpus",
         str(cpus),
-        "+command",
     ]
-
+    # Add in the interface if using TCP/IP
+    interface = kwargs.get("ifname",None)
+    if interface:
+        cmd.extend(["+ifname", interface])
+    cmd.append("+command")
     # collect DB binaries and libraries from the config
     db_cmd = [CONFIG.database_exe, CONFIG.database_conf, "--loadmodule", CONFIG.redisai]
+
     # add extra redisAI configurations
     for arg, value in rai_args.items():
         if value:
             # RAI wants arguments for inference in all caps
             # ex. THREADS_PER_QUEUE=1
             db_cmd.append(f"{arg.upper()} {str(value)}")
 
-    # add port and log information
-    db_cmd.extend(["--port", str(port), "--logfile", db_log])  # usually /dev/null
+    db_cmd.extend(["--port", str(kwargs["port"])])
+
+    # Add socket and permissions for UDS
+    unix_socket = kwargs.get("unix_socket", None)
+    socket_permissions = kwargs.get("socket_permissions", None)
+
+    if unix_socket and socket_permissions:
+        db_cmd.extend(
+            [
+                "--unixsocket", str(unix_socket),
+                "--unixsocketperm", str(socket_permissions)
+            ]
+        )
+    elif bool(unix_socket) ^ bool(socket_permissions):
+        raise SSInternalError(
+            "`unix_socket` and `socket_permissions` must both be defined or undefined."
+            )
+
+    db_cmd.extend(["--logfile", db_log])  # usually /dev/null
```

Comment:
Should `"/dev/null"` be the default value? Would I ever want to be able to extend this list with `["--logfile", None]`?

Maybe guard with a null check?
```py
if db_log is not None:
    db_cmd.extend(...)
```

Reply:
I'm actually not sure why this is a kwarg and not a required argument. If you look at the (one) reference to it; it is always specified. I'll do a quick refactor and check to make sure it doesn't break anything if I change it to an arg.